app.js

```diff
@@ -664,7 +665,26 @@ Promise.all([findCompilers(), aws.initConfig(awsProps)])
         logger.info("=======================================");
         webServer.use(Raven.errorHandler());
         webServer.on('error', err => logger.error('Caught error:', err, "(in web error handler; continuing)"));
-        webServer.listen(port, hostname);
+
+        const ss = systemdSocket();
+        if (ss) {
+            const timeout = (typeof process.env.IDLE_TIMEOUT !== 'undefined' ? process.env.IDLE_TIMEOUT : 300) * 1000; // ms (5 min default)
```

Comment:
Could you turn this into a function please.

Reply:
:+1: to this.